### PR TITLE
fix: order --resume <uuid> UUID misroute, redundant validation, and system prompt duplication

### LINKED ISSUE #1194: order --resume <uuid> fails: UUID misroute + redundant recipe validation + system prompt duplication
## Problem

`autoskillit order --resume 4b581974-1f19-4aec-8405-78c5ede5e233` fails with:

```
Recipe not found: '4b581974-1f19-4aec-8405-78c5ede5e233'
Available recipes:
  - implementation
  - implementation-groups
  - merge-prs
  - remediation
  - research
  - full-audit
  - smoke-test
```

Three layered problems prevent `order --resume <uuid>` from working correctly.

### Problem 1: CLI Argument Misroute

The `order` command signature (`src/autoskillit/cli/app.py:540`) is:

```python
def order(recipe: str | None = None, session_id: str | None = None, *, resume: bool = False):
```

- `recipe` — optional **positional** (1st position)
- `session_id` — optional **positional** (2nd position)
- `resume` — keyword-only **bool flag** (`--resume`, consumes 0 tokens)

When the user runs `autoskillit order --resume <uuid>`:
- `--resume` → `resume=True` (bool flag, does **not** consume the UUID)
- `<uuid>` → first positional → `recipe="4b581974-..."` (**wrong!**)
- `session_id` → remains `None`

Then at line 618, `find_recipe_by_name("4b581974-...", ...)` finds no match and the code exits before `resume_spec` is ever consumed.

Compare with `cook` (`src/autoskillit/cli/app.py:530`):

```python
def _cook_cmd(session_id: str | None = None, *, resume: bool = False):
```

`cook` has only one positional (`session_id`), so `cook --resume <uuid>` correctly routes the UUID. The asymmetry is that `order` has `recipe` occupying the first positional slot, intercepting the UUID.

### Problem 2: Recipe Validation Is Unconditional and Unnecessary on Resume

Even if the UUID were correctly routed to `session_id`, the `order` command still requires a recipe name and runs the full validation gauntlet. None of this is needed on resume — the session already has the recipe loaded, ingredients collected, and pipeline state intact.

| Step | Location | Needed on resume? |
|------|----------|-------------------|
| `find_recipe_by_name` + `load_recipe` + `validate_recipe` | app.py:618-644 | **No** — validated on original launch |
| Subset-disabled gate (interactive prompt) | app.py:646-669 | **No** — already answered on original launch |
| Pack gate (interactive prompt) | app.py:671-700 | **No** — already answered on original launch |
| `show_cook_preview` | app.py:705 | **No** — user already saw it |
| "Launch session?" confirmation | app.py:710-714 | Arguably yes for safety |
| `_build_orchestrator_prompt` + `--append-system-prompt` | app.py:716-718, _session_launch.py:36-43 | **No — harmful** (see Problem 3) |

Two existing code paths already prove resume works without recipe validation:

1. **`cook --resume <uuid>`** (`src/autoskillit/cli/_cook.py:94-174`): Supports resume with zero recipe validation, zero system prompt injection. Works fine.
2. **`order` open-kitchen path** (`src/autoskillit/cli/app.py:598-607`): When the user selects "Open kitchen (no recipe)" from the picker, `order` skips the entire recipe validation gauntlet and calls `_launch_cook_session` directly with `resume_spec`.

### Problem 3: System Prompt Duplication on Resume

`_run_interactive_session` (`src/autoskillit/cli/_session_launch.py:30-43`) unconditionally appends `--append-system-prompt <full_orchestrator_prompt>` to every claude invocation, including resumes. The resulting command is:

```
claude --dangerously-skip-permissions --resume <id>
       --tools AskUserQuestion
       --append-system-prompt "<full orchestrator prompt>"
```

On resume, Claude Code restores the original session's full conversation history (which already includes the system prompt from the first launch). The `--append-system-prompt` then **adds the orchestrator prompt a second time**, including "FIRST ACTION" instructions that tell Claude to re-run initialization steps (sleep 2, ToolSearch, open_kitchen, collect ingredients) — even if the session was mid-pipeline.

The `session_start_hook` (`src/autoskillit/hooks/session_start_hook.py`) already handles resume correctly: it detects non-empty transcripts, reads the recipe name from kitchen state, and injects a targeted `open_kitchen(name='<recipe>')` reminder. The full orchestrator prompt re-injection is redundant and harmful.

## Affected Components

- `src/autoskillit/cli/app.py:540` — `order` command definition and argument layout
- `src/autoskillit/cli/app.py:565-722` — entire recipe validation gauntlet (unconditional, no resume bypass)
- `src/autoskillit/cli/_session_launch.py:30-43` — unconditional `--append-system-prompt` injection
- `src/autoskillit/core/_type_resume.py:42-48` — `resume_spec_from_cli` (correct, but never reached due to Problem 1)
- `src/autoskillit/hooks/session_start_hook.py` — already handles resume correctly (not broken)

### Current (broken) flow

```
CLI tokens: ["order", "--resume", "4b581974-..."]
                         |                  |
                   resume=True       recipe="4b581974-..."  <-- WRONG
                                     session_id=None

→ _resume = True (line 569)
→ resume_spec = BareResume()  (line 570, session_id is None)
→ recipe is not None, so skip picker (line 576)
→ find_recipe_by_name("4b581974-...", cwd) → None (line 618)
→ "Recipe not found: '4b581974-...'" + sys.exit(1) (line 621)
```

### Expected flow

```
CLI tokens: ["order", "--resume", "4b581974-..."]

→ Detect UUID in recipe positional, reroute to session_id
→ _resume = True
→ resume_spec = NamedResume(session_id="4b581974-...")
→ Skip recipe validation (session already has it)
→ _launch_cook_session with minimal/no system prompt, resume_spec
→ session_start_hook handles open_kitchen reminder
```

## Historical Context

- **Commit `c0a1275` (#510)**: Added `--resume` flag to both `cook` and `order`. The original design required `order <recipe> --resume [<uuid>]` — recipe name was always required.
- **Commit `000ff42` (#548)**: Fixed `cook --resume <uuid>` by moving `session_id` from keyword-only to positional. This fix was specific to `cook` and did not change `order`.
- This is a **recurring pattern** — the positional argument design creates a parsing trap where users expect `--resume <value>` to consume the value, but because `--resume` is a bool flag, the value falls into the next available positional slot.

### Fix 1: UUID Detection and Reroute (`src/autoskillit/cli/app.py`, `order` function)

When `resume=True` and `recipe` matches UUID format and `session_id is None`, reroute `recipe` to `session_id` and set `recipe = None`. UUID pattern: `[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}`.

### Fix 2: Resume Fast-Path (`src/autoskillit/cli/app.py`, `order` function)

When `_resume` is true and `recipe` is None (either originally or after UUID reroute), skip the entire recipe validation gauntlet (find, load, validate, subset gate, pack gate, preview) and go directly to `_launch_cook_session`. The open-kitchen path at line 598-607 already demonstrates this pattern.

### Fix 3: Suppress System Prompt on Resume (`src/autoskillit/cli/_session_launch.py`)

When `resume_spec` is `BareResume` or `NamedResume`, do not append `--append-system-prompt` to the command. The session already has the orchestrator prompt in its restored history, and the `session_start_hook` handles the kitchen-reopen reminder. Re-injecting the full prompt causes duplication and "FIRST ACTION" confusion on mid-pipeline sessions.

### Fix 4: Tests

- `order --resume <uuid>` (no recipe name) reroutes UUID and launches successfully
- `order --resume` (bare, no UUID) enters resume without recipe validation
- Verify `--append-system-prompt` is suppressed when resume_spec is BareResume or NamedResume

## Test Gap Analysis

Current tests cover only the `order <recipe> --resume <uuid>` pattern (`tests/cli/test_cook.py:1469`). No test exists for `order --resume <uuid>` without a recipe name. No test covers the combined `--resume` + `--append-system-prompt` argv interaction — existing tests are siloed between `test_commands.py` (no `--append-system-prompt`) and `test_session_launch.py` (no resume).

## Dependencies

- **Blocked by:** None
- **Blocks:** #1064 (Scoped resume picker)
- **Why:** This issue fixes the foundational resume bugs (UUID misroute, redundant recipe validation, system prompt duplication) that #1064's filtered session picker depends on. The picker selects a session UUID and passes it to `order --resume <uuid>` — if the UUID misroute isn't fixed, the picker's output would still land in the `recipe` positional and fail. The system prompt suppression fix is also required so picker-resumed sessions don't get duplicate prompts.

### PR DESCRIPTION
**Pull request by Trecek**
## Summary

Three bugs prevent `autoskillit order --resume <uuid>` from working. This fix addresses all three in a single cohesive change:

1. **UUID Misroute** (`app.py:540`): When `--resume` is a bool flag, the UUID falls into the `recipe` positional. A UUID-detection reroute at the top of `order()` moves it to `session_id`.
2. **Redundant Recipe Validation** (`app.py:565-722`): Resume sessions already have the recipe loaded; the entire validation gauntlet must be bypassed when resuming without a recipe name.
3. **System Prompt Duplication** (`_session_launch.py:36-43`): `--append-system-prompt` is unconditionally appended to every claude invocation, including resumes. When resuming, Claude Code restores the original session history (which already has the system prompt); the second injection causes "FIRST ACTION" confusion. The system prompt must be suppressed when `resume_spec` is `BareResume` or `NamedResume`.

### Process Flow Diagram

```mermaid
%%{init: {'flowchart': {'nodeSpacing': 40, 'rankSpacing': 50, 'curve': 'basis'}}}%%
flowchart TB
    %% CLASS DEFINITIONS %%
    classDef terminal fill:#1a237e,stroke:#7986cb,stroke-width:2px,color:#fff;
    classDef stateNode fill:#004d40,stroke:#4db6ac,stroke-width:2px,color:#fff;
    classDef handler fill:#e65100,stroke:#ffb74d,stroke-width:2px,color:#fff;
    classDef phase fill:#6a1b9a,stroke:#ba68c8,stroke-width:2px,color:#fff;
    classDef detector fill:#b71c1c,stroke:#ef5350,stroke-width:2px,color:#fff;
    classDef output fill:#00695c,stroke:#4db6ac,stroke-width:2px,color:#fff;

    %% TERMINALS %%
    START([START])
    COMPLETE([COMPLETE])
    ERROR([ERROR])

    subgraph Entry ["● CLI Entry — app.order()"]
        direction TB
        EnvCheck{"CLAUDECODE<br/>set?"}
        ParseArgs["● Parse CLI Args<br/>━━━━━━━━━━<br/>recipe positional<br/>session_id, --resume"]
        MakeResumeSpec["resume_spec_from_cli()<br/>━━━━━━━━━━<br/>NoResume / BareResume /<br/>NamedResume"]
    end

    subgraph UUIDGate ["● UUID Detection Gate"]
        direction TB
        UUIDCheck{"_resume=True AND<br/>recipe looks like UUID AND<br/>session_id is None?"}
        Reroute["● Reroute UUID<br/>━━━━━━━━━━<br/>session_id = recipe<br/>recipe = None<br/>resume_spec = NamedResume(uuid)"]
    end

    subgraph ResumeShortCircuit ["Resume Short-Circuit"]
        direction TB
        ResumeNoRecipe{"_resume=True AND<br/>recipe is None?"}
        LaunchOpenKitchen["Launch Open Kitchen<br/>━━━━━━━━━━<br/>_launch_cook_session()<br/>resume_spec passed through<br/>recipe validation SKIPPED"]
    end

    subgraph RecipePath ["Recipe Validation Path"]
        direction TB
        NeedPicker{"recipe is None?<br/>(picker)"}
        ShowPicker["Show Recipe Picker<br/>━━━━━━━━━━<br/>list_recipes() + timed_prompt()"]
        FindRecipe["find_recipe_by_name()<br/>━━━━━━━━━━<br/>project + bundled search"]
        LoadValidate["load_recipe() + validate_recipe()<br/>━━━━━━━━━━<br/>YAML parse + schema validation"]
        SubsetGate{"Disabled subsets<br/>required?"}
        SubsetPrompt["Subset Prompt<br/>━━━━━━━━━━<br/>1=temp env override<br/>2=update config<br/>3=cancel"]
        PackGate{"Default-disabled<br/>packs required?"}
        PackPrompt["Pack Prompt<br/>━━━━━━━━━━<br/>1=temp env override<br/>2=update config<br/>3=cancel"]
        ConfirmLaunch["show_cook_preview() + Confirm<br/>━━━━━━━━━━<br/>timed_prompt() Enter/n"]
    end

    subgraph SessionLaunch ["● Session Launch — _run_interactive_session()"]
        direction TB
        BuildCmd["build_interactive_cmd()<br/>━━━━━━━━━━<br/>resume_spec → --resume flag<br/>env_extras merged"]
        PluginCheck{"plugin<br/>installed?"}
        AssembleCmd["Assemble Final CMD<br/>━━━━━━━━━━<br/>base cmd + plugin_flags<br/>+ --tools AskUserQuestion"]
        SysPromptGate{"● is_resume?<br/>BareResume or NamedResume?"}
        AppendSysPrompt["Append System Prompt<br/>━━━━━━━━━━<br/>--append-system-prompt<br/>orchestrator prompt"]
        SuppressSysPrompt["● Suppress System Prompt<br/>━━━━━━━━━━<br/>no --append-system-prompt<br/>prevents duplication on resume"]
        SubprocRun["subprocess.run(cmd, env)<br/>━━━━━━━━━━<br/>terminal_guard() context"]
    end

    %% MAIN FLOW %%
    START --> EnvCheck
    EnvCheck -->|"set"| ERROR
    EnvCheck -->|"clear"| ParseArgs
    ParseArgs --> MakeResumeSpec
    MakeResumeSpec --> UUIDCheck

    UUIDCheck -->|"yes — UUID fix"| Reroute
    UUIDCheck -->|"no"| ResumeNoRecipe
    Reroute --> ResumeNoRecipe

    ResumeNoRecipe -->|"yes — skip validation"| LaunchOpenKitchen
    ResumeNoRecipe -->|"no — has recipe or fresh"| NeedPicker
    LaunchOpenKitchen --> BuildCmd

    NeedPicker -->|"yes"| ShowPicker
    NeedPicker -->|"no"| FindRecipe
    ShowPicker --> FindRecipe
    FindRecipe -->|"not found"| ERROR
    FindRecipe -->|"found"| LoadValidate
    LoadValidate -->|"parse / schema error"| ERROR
    LoadValidate -->|"valid"| SubsetGate

    SubsetGate -->|"yes"| SubsetPrompt
    SubsetGate -->|"no"| PackGate
    SubsetPrompt -->|"1: temp env"| PackGate
    SubsetPrompt -->|"2: update config"| PackGate
    SubsetPrompt -->|"3: cancel"| COMPLETE

    PackGate -->|"yes"| PackPrompt
    PackGate -->|"no"| ConfirmLaunch
    PackPrompt -->|"1: temp env"| ConfirmLaunch
    PackPrompt -->|"2: update config"| ConfirmLaunch
    PackPrompt -->|"3: cancel"| COMPLETE

    ConfirmLaunch -->|"n / no"| COMPLETE
    ConfirmLaunch -->|"Enter: confirm"| BuildCmd

    BuildCmd --> PluginCheck
    PluginCheck -->|"not installed"| AssembleCmd
    PluginCheck -->|"installed"| AssembleCmd
    AssembleCmd --> SysPromptGate
    SysPromptGate -->|"yes — resume"| SuppressSysPrompt
    SysPromptGate -->|"no — fresh session"| AppendSysPrompt
    SuppressSysPrompt --> SubprocRun
    AppendSysPrompt --> SubprocRun
    SubprocRun -->|"returncode = 0"| COMPLETE
    SubprocRun -->|"returncode ≠ 0"| ERROR

    %% CLASS ASSIGNMENTS %%
    class START,COMPLETE,ERROR terminal;
    class EnvCheck,UUIDCheck,ResumeNoRecipe,NeedPicker,SubsetGate,PackGate,SysPromptGate detector;
    class ParseArgs,MakeResumeSpec,Reroute,LaunchOpenKitchen handler;
    class ShowPicker,FindRecipe,LoadValidate,SubsetPrompt,PackPrompt,ConfirmLaunch phase;
    class BuildCmd,AssembleCmd,AppendSysPrompt,SuppressSysPrompt,SubprocRun handler;
    class PluginCheck stateNode;
```

### State Lifecycle Diagram

```mermaid
%%{init: {'flowchart': {'nodeSpacing': 50, 'rankSpacing': 65, 'curve': 'basis'}}}%%
flowchart TB
    %% CLASS DEFINITIONS %%
    classDef cli fill:#1a237e,stroke:#7986cb,stroke-width:2px,color:#fff;
    classDef stateNode fill:#004d40,stroke:#4db6ac,stroke-width:2px,color:#fff;
    classDef handler fill:#e65100,stroke:#ffb74d,stroke-width:2px,color:#fff;
    classDef phase fill:#6a1b9a,stroke:#ba68c8,stroke-width:2px,color:#fff;
    classDef output fill:#00695c,stroke:#4db6ac,stroke-width:2px,color:#fff;
    classDef detector fill:#b71c1c,stroke:#ef5350,stroke-width:2px,color:#fff;
    classDef gap fill:#ff6f00,stroke:#ffa726,stroke-width:2px,color:#000;
    classDef terminal fill:#1a237e,stroke:#7986cb,stroke-width:2px,color:#fff;

    START(["CLI: autoskillit order"]):::terminal

    subgraph CLIArgs ["CLI ARGUMENT FIELDS"]
        direction LR
        recipe_arg["MUTABLE: recipe<br/>━━━━━━━━━━<br/>Recipe name OR UUID<br/>Cleared on reroute"]
        session_id_arg["MUTABLE: session_id<br/>━━━━━━━━━━<br/>Explicit session UUID<br/>Set from recipe on reroute"]
        resume_flag["DERIVED: _is_resume<br/>━━━━━━━━━━<br/>resume OR session_id≠None<br/>Read-only after compute"]
    end

    subgraph Gate1 ["GATE 1 — CLAUDECODE GUARD (INIT_ONLY)"]
        env_guard["CLAUDECODE env check<br/>━━━━━━━━━━<br/>Blocks execution inside<br/>active Claude session<br/>FAIL-FAST → exit 1"]
    end

    subgraph Gate2 ["● GATE 2 — UUID REROUTE (STATE MUTATION)"]
        uuid_check{"_is_resume AND<br/>recipe≠None AND<br/>session_id=None AND<br/>recipe matches UUID?"}
        uuid_reroute["MUTATE STATE<br/>━━━━━━━━━━<br/>session_id ← recipe<br/>recipe ← None<br/>resume_spec ← NamedResume(session_id)"]
    end

    subgraph Gate3 ["● GATE 3 — RESUME BYPASS (VALIDATION SKIP)"]
        resume_bypass{"_is_resume AND<br/>recipe=None?"}
        skip_validation["SKIP VALIDATION<br/>━━━━━━━━━━<br/>Launch directly via<br/>_launch_cook_session<br/>No recipe lookup"]
    end

    subgraph Gate4 ["GATE 4 — RECIPE VALIDATION"]
        recipe_lookup["find_recipe_by_name<br/>━━━━━━━━━━<br/>FAIL → exit 1"]
        yaml_parse["load_recipe (YAML)<br/>━━━━━━━━━━<br/>YAMLError → exit 1<br/>ValueError → exit 1"]
        schema_valid["validate_recipe<br/>━━━━━━━━━━<br/>errors → exit 1"]
    end

    subgraph Gate5 ["GATE 5 — SUBSET / PACK GATE"]
        subset_gate["Subset-disabled check<br/>━━━━━━━━━━<br/>Prompt: temp/perm/cancel"]
        pack_gate["Pack-disabled check<br/>━━━━━━━━━━<br/>Prompt: temp/perm/cancel"]
    end

    subgraph ResumeSpec ["INIT_ONLY: resume_spec (discriminated union)"]
        no_resume["NoResume<br/>━━━━━━━━━━<br/>Fresh session<br/>No --resume flag"]
        bare_resume["BareResume<br/>━━━━━━━━━━<br/>resume=True, no UUID<br/>Bare --resume flag"]
        named_resume["NamedResume(session_id)<br/>━━━━━━━━━━<br/>Explicit UUID<br/>--resume <uuid> flag"]
    end

    subgraph SessionLaunch ["● _run_interactive_session (SYSTEM PROMPT GATE)"]
        is_resume_check{"isinstance(resume_spec,<br/>(BareResume, NamedResume))?"}
        suppress_prompt["INIT_PRESERVE: system_prompt<br/>━━━━━━━━━━<br/>Omit --append-system-prompt<br/>Resume: keep prior context"]
        inject_prompt["INIT_PRESERVE: system_prompt<br/>━━━━━━━━━━<br/>Inject --append-system-prompt<br/>Fresh: install instructions"]
    end

    CMD_OUT(["subprocess: claude <flags>"]):::terminal

    %% FLOW %%
    START --> recipe_arg & session_id_arg & resume_flag
    recipe_arg & session_id_arg & resume_flag --> env_guard

    env_guard -->|"PASS"| uuid_check
    env_guard -->|"FAIL"| BLOCKED(["exit 1"]):::terminal

    uuid_check -->|"YES"| uuid_reroute
    uuid_reroute --> resume_bypass
    uuid_check -->|"NO"| resume_bypass

    resume_bypass -->|"YES"| skip_validation
    resume_bypass -->|"NO"| recipe_lookup

    recipe_lookup --> yaml_parse --> schema_valid --> subset_gate --> pack_gate

    skip_validation --> no_resume & bare_resume & named_resume
    pack_gate --> no_resume & bare_resume & named_resume

    no_resume --> is_resume_check
    bare_resume --> is_resume_check
    named_resume --> is_resume_check

    is_resume_check -->|"BareResume or NamedResume"| suppress_prompt
    is_resume_check -->|"NoResume"| inject_prompt

    suppress_prompt --> CMD_OUT
    inject_prompt --> CMD_OUT

    %% CLASS ASSIGNMENTS %%
    class recipe_arg,session_id_arg stateNode;
    class resume_flag phase;
    class env_guard detector;
    class uuid_check,resume_bypass phase;
    class uuid_reroute gap;
    class recipe_lookup,yaml_parse,schema_valid detector;
    class subset_gate,pack_gate handler;
    class no_resume,bare_resume,named_resume cli;
    class is_resume_check phase;
    class suppress_prompt gap;
    class inject_prompt output;
    class skip_validation output;
    class START,BLOCKED,CMD_OUT terminal;
```

### Scenarios Diagram

```mermaid
%%{init: {'flowchart': {'nodeSpacing': 40, 'rankSpacing': 55, 'curve': 'basis'}}}%%
flowchart LR
    %% CLASS DEFINITIONS %%
    classDef cli fill:#1a237e,stroke:#7986cb,stroke-width:2px,color:#fff;
    classDef stateNode fill:#004d40,stroke:#4db6ac,stroke-width:2px,color:#fff;
    classDef handler fill:#e65100,stroke:#ffb74d,stroke-width:2px,color:#fff;
    classDef phase fill:#6a1b9a,stroke:#ba68c8,stroke-width:2px,color:#fff;
    classDef output fill:#00695c,stroke:#4db6ac,stroke-width:2px,color:#fff;
    classDef detector fill:#b71c1c,stroke:#ef5350,stroke-width:2px,color:#fff;
    classDef terminal fill:#1a237e,stroke:#7986cb,stroke-width:2px,color:#fff;

    %% ================================================================
    %% SCENARIO 1: Fresh order <recipe> — happy path
    %% ================================================================
    subgraph S1 ["SCENARIO 1: Fresh order &lt;recipe&gt; — system prompt injected"]
        direction LR
        S1_CLI["● app.order<br/>━━━━━━━━━━<br/>recipe='my-recipe'<br/>resume=False"]
        S1_VAL["validate_recipe<br/>━━━━━━━━━━<br/>load_recipe + validate<br/>subset/pack gates"]
        S1_PROMPT["_build_orchestrator_prompt<br/>━━━━━━━━━━<br/>recipe name + MCP prefix<br/>+ ingredients table"]
        S1_LAUNCH["_launch_cook_session<br/>━━━━━━━━━━<br/>resume_spec=NoResume"]
        S1_SESSION["● _run_interactive_session<br/>━━━━━━━━━━<br/>NoResume → appends<br/>--append-system-prompt"]
        S1_CLAUDE["claude subprocess<br/>━━━━━━━━━━<br/>--tools AskUserQuestion<br/>--append-system-prompt &lt;prompt&gt;"]
    end

    S1_CLI -->|"finds recipe"| S1_VAL
    S1_VAL -->|"valid"| S1_PROMPT
    S1_PROMPT -->|"prompt built"| S1_LAUNCH
    S1_LAUNCH -->|"NoResume"| S1_SESSION
    S1_SESSION -->|"system prompt included"| S1_CLAUDE

    %% ================================================================
    %% SCENARIO 2: order --resume <uuid> — UUID misroute fix
    %% ================================================================
    subgraph S2 ["SCENARIO 2: order --resume &lt;uuid&gt; — UUID rerouted, prompt suppressed"]
        direction LR
        S2_CLI["● app.order<br/>━━━━━━━━━━<br/>recipe='4b581974-...'<br/>resume=True, session_id=None"]
        S2_DETECT["● UUID detector<br/>━━━━━━━━━━<br/>_UUID_RE.match(recipe)<br/>→ session_id=UUID, recipe=None"]
        S2_RESUME["resume_spec_from_cli<br/>━━━━━━━━━━<br/>NamedResume(session_id=UUID)"]
        S2_LAUNCH["_launch_cook_session<br/>━━━━━━━━━━<br/>resume_spec=NamedResume"]
        S2_SESSION["● _run_interactive_session<br/>━━━━━━━━━━<br/>NamedResume → suppresses<br/>--append-system-prompt"]
        S2_CLAUDE["claude subprocess<br/>━━━━━━━━━━<br/>--resume &lt;uuid&gt;<br/>--tools AskUserQuestion"]
    end

    S2_CLI -->|"UUID in recipe arg"| S2_DETECT
    S2_DETECT -->|"rerouted to session_id"| S2_RESUME
    S2_RESUME -->|"NamedResume"| S2_LAUNCH
    S2_LAUNCH -->|"NamedResume"| S2_SESSION
    S2_SESSION -->|"NO system prompt"| S2_CLAUDE

    %% ================================================================
    %% SCENARIO 3: order <recipe> --resume <uuid> — named resume
    %% ================================================================
    subgraph S3 ["SCENARIO 3: order &lt;recipe&gt; --resume &lt;uuid&gt; — validates recipe, prompt suppressed"]
        direction LR
        S3_CLI["● app.order<br/>━━━━━━━━━━<br/>recipe='my-recipe'<br/>session_id='fa910a41-...'"]
        S3_VAL["validate_recipe<br/>━━━━━━━━━━<br/>recipe found + validated<br/>(name not a UUID)"]
        S3_LAUNCH["_launch_cook_session<br/>━━━━━━━━━━<br/>resume_spec=NamedResume"]
        S3_SESSION["● _run_interactive_session<br/>━━━━━━━━━━<br/>isinstance NamedResume<br/>→ suppresses system prompt"]
        S3_CLAUDE["claude subprocess<br/>━━━━━━━━━━<br/>--resume fa910a41-...<br/>--tools AskUserQuestion"]
    end

    S3_CLI -->|"non-UUID recipe name"| S3_VAL
    S3_VAL -->|"valid"| S3_LAUNCH
    S3_LAUNCH -->|"NamedResume"| S3_SESSION
    S3_SESSION -->|"NO system prompt"| S3_CLAUDE

    %% ================================================================
    %% SCENARIO 4: order --resume (bare) — skips recipe validation
    %% ================================================================
    subgraph S4 ["SCENARIO 4: order --resume (bare) — open kitchen, prompt suppressed"]
        direction LR
        S4_CLI["● app.order<br/>━━━━━━━━━━<br/>recipe=None<br/>resume=True"]
        S4_BARERESUME["resume_spec_from_cli<br/>━━━━━━━━━━<br/>BareResume()<br/>no recipe lookup"]
        S4_LAUNCH["_launch_cook_session<br/>━━━━━━━━━━<br/>empty system_prompt<br/>resume_spec=BareResume"]
        S4_SESSION["● _run_interactive_session<br/>━━━━━━━━━━<br/>isinstance BareResume<br/>→ suppresses system prompt"]
        S4_CLAUDE["claude subprocess<br/>━━━━━━━━━━<br/>bare --resume<br/>--tools AskUserQuestion"]
    end

    S4_CLI -->|"resume=True, no recipe"| S4_BARERESUME
    S4_BARERESUME -->|"BareResume"| S4_LAUNCH
    S4_LAUNCH -->|"BareResume"| S4_SESSION
    S4_SESSION -->|"NO system prompt"| S4_CLAUDE

    %% CLASS ASSIGNMENTS %%
    class S1_CLI,S2_CLI,S3_CLI,S4_CLI cli;
    class S1_VAL,S3_VAL phase;
    class S1_PROMPT handler;
    class S2_DETECT detector;
    class S2_RESUME,S4_BARERESUME stateNode;
    class S1_LAUNCH,S2_LAUNCH,S3_LAUNCH,S4_LAUNCH handler;
    class S1_SESSION,S2_SESSION,S3_SESSION,S4_SESSION handler;
    class S1_CLAUDE,S2_CLAUDE,S3_CLAUDE,S4_CLAUDE output;
```

Closes #1194

## Implementation Plan

Plan file: `/home/talon/projects/autoskillit-runs/impl-20260424-192339-259031/.autoskillit/temp/make-plan/order_resume_uuid_fix_plan_2026-04-24_000000.md`

🤖 Generated with [Claude Code](https://claude.com/claude-code) via AutoSkillit

## Token Usage Summary

| Step | uncached | output | cache_read | cache_write | count | time |
|------|----------|--------|------------|-------------|-------|------|
| retry_worktree | 108 | 3.1k | 452.0k | 34.9k | 1 | 2m 24s |
| prepare_pr | 52 | 4.9k | 188.0k | 35.3k | 1 | 1m 28s |
| run_arch_lenses | 140 | 17.2k | 472.6k | 143.0k | 2 | 6m 6s |
| **Total** | 300 | 25.2k | 1.1M | 213.2k | | 10m 0s |

### CODE REVIEW

File: src/autoskillit/cli/app.py [hunk 1]
```diff
@@ -569,6 +570,22 @@ def order(recipe: str | None = None, session_id: str | None = None, *, resume: b
     _resume = resume or (session_id is not None)
     resume_spec = resume_spec_from_cli(resume=_resume, session_id=session_id)
 
+    _UUID_RE = re.compile(r"^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$", re.IGNORECASE)
```
Comment (Trecek) [inline src/autoskillit/cli/app.py]: [warning] defense: _UUID_RE is compiled inside the `order` function body on every invocation. Move to module-level constant to avoid repeated compilation and establish it as a named boundary validator.

File: src/autoskillit/cli/app.py [hunk 2]
```diff
@@ -569,6 +570,22 @@ def order(recipe: str | None = None, session_id: str | None = None, *, resume: b
...
+    if _resume and recipe is not None and session_id is None and _UUID_RE.match(recipe):
```
Comment (Trecek) [inline src/autoskillit/cli/app.py:575]: [warning] defense: UUID reroute guard only fires when `session_id is None`. If a caller passes both `--resume` and an explicit `--session-id` together with a UUID-shaped recipe name, the reroute is silently skipped and the UUID is forwarded to recipe lookup. Document or handle the `session_id is not None` case explicitly.

File: tests/cli/test_cook.py [hunk 3]
```diff
@@ -1498,3 +1498,70 @@ def fake_launch(prompt, *, initial_message=None, extra_env=None, resume_spec=NoR
...
+
+        monkeypatch.delenv("CLAUDECODE", raising=False)
+        monkeypatch.chdir(tmp_path)
+
+        find_called = []
+        monkeypatch.setattr(
+            "autoskillit.recipe.find_recipe_by_name",
+            lambda *a, **kw: find_called.append(a) or None,
+        )
+
```
Comment (Trecek) [inline tests/cli/test_cook.py:1562]: [warning] tests: `pytest.raises(SystemExit)` without asserting `exc_info.value.code`. Sibling tests assert `code == 0`, so an unexpected error-path exit (e.g. `sys.exit(1)`) would still pass this test and hide a regression. Add `assert exc_info.value.code == 0`.

**Review by Trecek — COMMENTED** [review]
AutoSkillit PR Review — Verdict: approved_with_comments

**Review by Trecek — COMMENTED** [review]
AutoSkillit review: warning-only findings detected. See inline comments — no blocking changes required.

File: src/autoskillit/cli/app.py (hunk 1 above)
Comment (Trecek) [reply src/autoskillit/cli/app.py]: Agreed — fixed in 7c796bf6. _UUID_RE promoted to module-level constant (line 46 of app.py) so it is compiled once at import time, not on every order() invocation.

File: src/autoskillit/cli/app.py (hunk 2 above)
Comment (Trecek) [reply src/autoskillit/cli/app.py:575]: Valid observation — flagged for design decision. The session_id is None guard is intentional: when --session-id is already provided explicitly, no reroute is needed. However, the behavior when both --session-id and a UUID-shaped recipe positional are provided is undocumented. This edge case needs a documented decision (inline comment or docstring) before it can be called correct or incorrect.

File: tests/cli/test_cook.py (hunk 3 above)
Comment (Trecek) [reply tests/cli/test_cook.py:1562]: Agreed — fixed in 28b0b29d. Added 'with pytest.raises(SystemExit) as exc_info' and 'assert exc_info.value.code == 0' to match the three sibling tests in the same class (lines 1494-1496, 1518-1520, 1542-1544) that all guard the same exit path.
